plugins/radar-android-google-sleep/src/main/java/org.radarbase.passive.google.sleep/GoogleSleepManager.kt

```diff
@@ -116,12 +124,17 @@ class GoogleSleepManager(context: GoogleSleepService) : AbstractSourceManager<Go
     }
 
     private fun createSleepPendingIntent(): PendingIntent {
-        val intent = Intent(service, SleepReceiver::class.java)
+        val intent = Intent(ACTION_SLEEP_DATA)
         logger.info("Sleep pending intent created")
         return PendingIntent.getBroadcast(service, SLEEP_DATA_REQUEST_CODE, intent,
             PendingIntent.FLAG_CANCEL_CURRENT.toPendingIntentFlag(true))
     }
 
+    private fun unRegisterSleepReceiver() {
+        service.unregisterReceiver(sleepBroadcastReceiver)
```

Comment:
Add a try/catch `IllegalArgumentException` around this, if for any reason unregistering fails.